hybrid/decomposers.py

```diff
@@ -422,6 +428,91 @@ def next(self, state, **runopts):
 
         return state.updated(subproblem=subbqm)
 
+class SublatticeDecomposer(traits.ProblemDecomposer, traits.SISO, Runnable):
+    """Selects a lattice structured subproblem creating associated 
+    ``subproblem``, and ``embedding`` state fields.
+    This decomposer uses the state fields ``bqm``, ``origin_embeddings`` and 
+    ``problem_dims`` which must be compatible (in particular, through geometric
+    variable keying). 
+    The decomposer also uses the optional state fields ``exclude_dimensions``, 
+    ``geometric_offset`` and ``origin_embedding_index``.
+    If ``geometric_offset`` and/or ``origin_embedding_index`` are not state 
+    variables, these are assigned uniformly at random, the random number 
+    generator can be initialized with the class variable ``seed``. This creates 
+    workflows in which sublattices and embeddings are chosen independently and 
+    uniformly at random on every block. More complicated workflows might 
```

Comment:
A more powerful workflow would use information about the current energy, and recent energy changes. It would use information about the previous geometric position of updates. This is state history, the state variables (fields) on previous iterations. There are innumerable different ways to do this, for this pull request only the simplest case is considered (no consideration of state history)

Reply:
This is not the appropriate place for a comment like this. It relates to workflow strategy and not properties of this class. Removed